esque/ruleparser/expressionelement.py

```diff
@@ -0,0 +1,220 @@
+            if integer_operands:
+                return op1_converted % op2_converted
+            else:
+                raise TypeError("Remainder can only be used with integer arguments")
+        elif self.generic_operator.name == Operator.OPERATORS["BINARY_AND"].name:
+            if integer_operands:
+                return op1_converted & op2_converted
+            else:
+                raise TypeError("Binary operators can only be used with integer arguments")
+        elif self.generic_operator.name == Operator.OPERATORS["BINARY_OR"].name:
+            if integer_operands:
+                return op1_converted | op2_converted
+            else:
+                raise TypeError("Binary operators can only be used with integer arguments")
+        elif self.generic_operator.name == Operator.OPERATORS["BINARY_XOR"].name:
+            if integer_operands:
+                return op1_converted ^ op2_converted
+            else:
+                raise TypeError("Binary operators can only be used with integer arguments")
+        elif self.generic_operator.name == Operator.OPERATORS["LOGICAL_AND"].name:
+            if boolean_operands:
+                return op1_converted and op2_converted
+            else:
+                raise TypeError("Logical operators can only be used with boolean values")
+        elif self.generic_operator.name == Operator.OPERATORS["LOGICAL_OR"].name:
+            if boolean_operands:
+                return op1_converted or op2_converted
+            else:
+                raise TypeError("Logical operators can only be used with boolean values")
+        elif self.generic_operator.name == Operator.OPERATORS["LOGICAL_XOR"].name:
+            if boolean_operands:
+                return op1_converted ^ op2_converted
+            else:
+                raise TypeError("Logical operators can only be used with boolean values")
+
+
+class ComparisonBinaryOperator(AbstractBinaryOperator):
+    def evaluate(self, operand1: str, operand2: str):
+        op1_converted = None
+        op2_converted = None
+        if h.is_boolean(operand1) and h.is_boolean(operand2):
+            op1_converted = h.to_boolean(operand1)
+            op2_converted = h.to_boolean(operand2)
+        if h.is_float(operand1) and h.is_float(operand2):
+            # if they are numeric, it doesn't matter if they are integer or not
+            op1_converted = float(operand1)
+            op2_converted = float(operand2)
+        elif (h.is_date_string(operand1) or h.to_date_time(operand1)) and (
+            h.is_date_string(operand2) or h.is_date_time_string(operand2)
+        ):
+            # maybe they are date strings
+            op1_converted = h.to_date_time(operand1)
+            op2_converted = h.to_date_time(operand2)
+        else:
+            # otherwise, we'll test them as strings
+            op1_converted = operand1
+            op2_converted = operand2
+        if op1_converted is None or op2_converted is None:
```

Comment:
this can't happen

Reply:
The refactoring should have solved this.